pkg/sql/executor_statement_metrics.go

```diff
@@ -255,6 +255,10 @@ func (ex *connExecutor) recordStatementSummary(
 		}
 	}
 
+	if stmtFingerprintID != 0 {
+		ex.statsCollector.ObserveStatement(stmtFingerprintID, recordedStmtStats)
```

Comment:
In retrospect, should we rename `ObserveStatement` to be more insights-specific? It doesn't affect other statement stuff right?

I think it's confusing with `RecordStatement` above.

Reply:
Do we need the conditional here, can't the method handle the 0 ID?

Reply:
Additionally, why can't `RecordStatement` call `ObserveStatement`?

Reply:
This PR aims to separate insights and sql stats writing and we accomplished this by moving ObserveStatement to this layer. These ops can occur independently whereas they were being coupled before (with insights being scrapped if we don't record sql stats).